### PR TITLE
[CLOUDGA-22303] Add support for integration type PROMETHEUS

### PR DESCRIPTION
**Pull request by bhupendray-yb**
## Resource Definition
```terraform
resource "ybm_integration" "prometheus" {
  config_name = "prometheus-example"
  type        = "PROMETHEUS"
  prometheus_spec = {
    endpoint = "http://yourcompany.com/api/v1/otlp"
  }
}
```

## Terraform Plan
```shell
> tfp

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # ybm_integration.prometheus will be created
  + resource "ybm_integration" "prometheus" {
      + account_id      = (known after apply)
      + config_id       = (known after apply)
      + config_name     = "prometheus-example"
      + is_valid        = (known after apply)
      + project_id      = (known after apply)
      + prometheus_spec = {
          + endpoint = "http://yourcompany.com/api/v1/otlp"
        }
      + type            = "PROMETHEUS"
    }

Plan: 1 to add, 0 to change, 0 to destroy.
```

## Terraform Apply
```shell
> tfa -auto-approve

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # ybm_integration.prometheus will be created
  + resource "ybm_integration" "prometheus" {
      + account_id      = (known after apply)
      + config_id       = (known after apply)
      + config_name     = "prometheus-example"
      + is_valid        = (known after apply)
      + project_id      = (known after apply)
      + prometheus_spec = {
          + endpoint = "http://yourcompany.com/api/v1/otlp"
        }
      + type            = "PROMETHEUS"
    }

Plan: 1 to add, 0 to change, 0 to destroy.
ybm_integration.prometheus: Creating...
ybm_integration.prometheus: Creation complete after 3s

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

## Terraform State
```shell
> tf state show ybm_integration.prometheus
# ybm_integration.prometheus:
resource "ybm_integration" "prometheus" {
    account_id      = "3b98d883-1b63-40ce-a4d7-8dc2d6e0ba53"
    config_id       = "4200c8df-2fb6-4098-9239-b9f8b046d6c5"
    config_name     = "prometheus-example"
    is_valid        = true
    project_id      = "bebbcb6b-3927-4ac3-81ce-23bba0c75a2e"
    prometheus_spec = {
        endpoint = "http://yourcompany.com/api/v1/otlp"
    }
    type            = "PROMETHEUS"
}
```

## Terraform Destroy
```shell
> tf destroy -auto-approve
ybm_integration.prometheus: Refreshing state...

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  - destroy

Terraform will perform the following actions:

  # ybm_integration.prometheus will be destroyed
  - resource "ybm_integration" "prometheus" {
      - account_id      = "3b98d883-1b63-40ce-a4d7-8dc2d6e0ba53" -> null
      - config_id       = "0ccc8e03-5742-4fac-92e8-443f4f233017" -> null
      - config_name     = "prometheus-example" -> null
      - is_valid        = true -> null
      - project_id      = "bebbcb6b-3927-4ac3-81ce-23bba0c75a2e" -> null
      - prometheus_spec = {
          - endpoint = "http://yourcompany.com/api/v1/otlp" -> null
        } -> null
      - type            = "PROMETHEUS" -> null
    }

Plan: 0 to add, 0 to change, 1 to destroy.
ybm_integration.prometheus: Destroying...
ybm_integration.prometheus: Destruction complete after 1s
```

### CODE REVIEW

**Review by Siddarth-Baldwa — COMMENTED** [review]
Can we keep prometheus integration under FF?

**Comment by bansal01yash** [comment]
> Can we keep prometheus integration under FF?

There's no reason to, it is just creation of integration and we have confirmed with QA it can be tested before next release

**Review by Siddarth-Baldwa — APPROVED** [review]

File: docs/resources/integration.md [hunk 1]
```diff
@@ -116,6 +117,14 @@ Required:
 - `zone` (String) Grafana Zone.
 
 
+<a id="nestedatt--prometheus_spec"></a>
+### Nested Schema for `prometheus_spec`
+
+Required:
+
+- `endpoint` (String) Prometheus OTLP endpoint URL
```
Comment (bansal01yash) [inline docs/resources/integration.md:125]: `Prometheus OTLP endpoint URL of the form http://my-prometheus-endpoint/api/v1/otlp`
Perhaps?

File: managed/resource_integration.go [hunk 2]
```diff
@@ -80,6 +80,18 @@ func (r resourceIntegrationType) getSchemaAttributes() map[string]tfsdk.Attribut
 				},
 			}),
 		},
+		"prometheus_spec": {
+			Description: "The specifications of a Prometheus integration.",
+			Optional:    true,
+			Validators:  onlyContainsPath("prometheus_spec"),
+			Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
+				"endpoint": {
+					Description: "Prometheus OTLP endpoint URL",
```
Comment (bansal01yash) [inline managed/resource_integration.go:89]: Same here

File: docs/resources/integration.md (hunk 1 above)
Comment (bhupendray-yb) [reply docs/resources/integration.md]: fixed.

**Pull request merged**